internal/provisioner/README.md

```diff
@@ -1,6 +1,60 @@
-# provisioner
+# Provisioners
 
 The following subdirectories contain concrete Provisioner implementations.
 
 - [plain](plain/README.md) - provisions `plain+v0` k8s bundles
 - [registry](registry/README.md) - provisions `registry+v1` OLM bundles
+
+## Global Provisioner Concepts
+Below is an example of the provisioner reconciliation flow:
+
+```mermaid
+graph TD
+    C[Provisioner]
+    C -->|Watches| D[Bundle]
+    C -->|Watches| E[BundleInstance]
+    D -->|References| F[Content]
+    E -->|Creates/Manages| F[Content]
+```
+### Make bundle content available but do not install it
```

Comment:
I think starting with this section is a bit confusing to users -- is inspecting bundle content without applying it the first provisioner concept we want to show off? The plain doc had the use case of installing a Bundle via a BI as the primary use case, I think it's more intuitive.

Reply:
That section looks really good. I would argue that we should move `Make bundle content available but do not install it` down even further and move `Pivoting between bundle versions` after the new section, since that concept is more important. This was just an oversight in the original doc I wrote.

Reply:
Gotcha, I swapped those two now!